### PR TITLE
Reimplement contributor form

### LINKED ISSUE #1673: Reimplement contributor form & emails
* Need to reimplement a contributor form that can be filled out after signup
* We used to have a lambda handler that would send an email to open a ticket when a user filled in the contributor form. Need to reimplement those too.
* Import past forms (that can be matched to a user)

@ItsiW: when do you want signup emails? Always? Only when they fill in some of the feedback? Only when they say they'd like to contribute?

### PR DESCRIPTION
**Pull request by aapeliv**
So that users can sign up to volunteer again, or fill the form in again if they change their mind.

Closes #1673.

I hope I didn't butcher the frontend too bad :/

Regarding the duplicate of state of whether the form is filled in the backend: this is a bit of debt from the way the forms were done in the past. I think we should import the existing forms (that can be matched against emails/usernames) and then get rid of the semi-duplication of state. But that's another PR.

**Backend checklist**
- [x] Formatted my code by running `autoflake -r -i --remove-all-unused-imports src && isort . && black .` in `app/backend`
- [ ] Added tests for any new code or added a regression test if fixing a bug
- [ ] All tests pass
- [x] Run the backend locally and it works
- [ ] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

**Frontend checklist**
- [x] Formatted my code with `yarn format && yarn lint --fix`
- [x] There are no warnings from `yarn lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [x] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

File: app/proto/account.proto [hunk 1]
```diff
@@ -93,17 +94,10 @@ message ChangeEmailReq {
...
-  string email = 4;
-  uint32 age = 5;
-  string gender = 6;
-  string location = 7;
 }
 
-message MarkContributorFormFilledReq {
-  bool filled_contributor_form = 1;
+message FillContributorFormReq {
+  org.couchers.auth.ContributorForm contributor_form = 1;
```
Comment (kalvdans) [inline app/proto/account.proto:100]: It this extra indirection to be backwards compatible for future expansions?

**Review by kalvdans — DISMISSED** [review]
Backend looks fine, just a small note.

File: app/proto/account.proto (hunk 1 above)
Comment (aapeliv) [reply app/proto/account.proto:100]: Ah, yes; I was going to mention that. I've added it into the list of things to think about for API refactoring: #899, but decided to do it this way for now.

File: app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        queryClient.invalidateQueries(contributorFormInfoQueryKey);
+      },
+    }
+  );
+
+  const handleSubmit = async (form: ContributorFormPb.AsObject) => {
+    mutate(form);
+  };
+
+  return isLoading || queryLoading ? (
```
Comment (lucaslcode) [inline app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx]:
```suggestion
  return queryLoading ? (
```

File: app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    {
+      onSuccess: async () => {
+        queryClient.invalidateQueries(contributorFormInfoQueryKey);
+      },
+    }
+  );
+
+  const handleSubmit = async (form: ContributorFormPb.AsObject) => {
+    mutate(form);
+  };
```
Comment (lucaslcode) [inline app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx]:
```suggestion
  const handleSubmit = async (form: ContributorFormPb.AsObject) => {
    await service.account.fillContributorForm(form);
    queryClient.invalidateQueries(contributorFormInfoQueryKey);
    setFillState("success")'
  };
```
Error handling and loading state is all handled in the form

File: app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  return isLoading || queryLoading ? (
+    <CircularProgress />
+  ) : (
+    <>
+      {error && <Alert severity="error">{error.message}</Alert>}
```
Comment (lucaslcode) [inline app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx]:
```suggestion
```

File: app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      {queryError && <Alert severity="error">{queryError?.message}</Alert>}
+      {data && data.filledContributorForm && !fillAgain ? (
```
Comment (lucaslcode) [inline app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx]:
```suggestion
      {data?.filledContributorForm && !fillAgain ? (
```

File: app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import { useState } from "react";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { service } from "service";
+
+import { ALREADY_FILLED_IN, FILL_IN_AGAIN, SUCCESS_MSG } from "./constants";
+
+export default function StandaloneContributorForm() {
+  const queryClient = useQueryClient();
+
+  const [fillAgain, setFillAgain] = useState(false);
```
Comment (lucaslcode) [inline app/frontend/src/components/ContributorForm/StandaloneContributorForm.tsx]:
```suggestion
  const [fillState, setFillState] = useState<undefined | "filled" | "success" | "fillAgain">(undefined);
```

**Review by lucaslcode — CHANGES_REQUESTED** [review]

**Comment by aapeliv** [comment]
@lucaslcode tried to fix it up, not sure if I screwed it up?

**Comment by aapeliv** [comment]
Tested locally, works fine

**Review by lucaslcode — APPROVED** [review]

**Pull request merged**